### PR TITLE
08 周河 lesson-5

### PR DESCRIPTION
**Pull request by syuukawa**
08 周河 lesson-5

### CODE REVIEW

**Comment by syuukawa** [comment]
08 周河 lesson-5

File: projects/lesson-5/runtime/src/kitties.rs [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+/////设计加密猫模块 V4
+/////pub struct Kitty(
+/////   pub [u8; 16],
+/////   price: i32
+/////);
+/////
+/////给⾃己的小猫设定价钱
+/////pub fn set_price(origin, kitty_id: T::KittyIndex,price: i32)
+/////购买⼩猫
+/////pub fn buy_kitty(origin, to: T::AccountId, kitty_id: T::KittyIndex,price: i32)
```
Comment (kaichaosun) [inline projects/lesson-5/runtime/src/kitties.rs]: Add draft code is good, hope you implement it later.

File: projects/lesson-5/runtime/src/kitties.rs [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+
+		// 作业：实现 transfer(origin, to: T::AccountId, kitty_id: T::KittyIndex)
+		// 使用 ensure! 来保证只有主人才有权限调用 transfer
+		// 使用 OwnedKitties::append 和 OwnedKitties::remove 来修改小猫的主人
+		pub fn transfer(origin, to: T::AccountId, kitty_id: T::KittyIndex){
+			let sender = ensure_signed(origin)?;
+			// TODO-001
+//			ensure!(<OwnedKitties<T>>::get((sender.clone(),kitty_id)) != Option<KittyLinkedItem{None,None}>,"kitty_id is not owned by the sender");
+			<OwnedKitties<T>>::append(&to,kitty_id);
+			<OwnedKitties<T>>::remove(&sender,kitty_id);
```
Comment (kaichaosun) [inline projects/lesson-5/runtime/src/kitties.rs]: You miss setting a few storage like `KittyOwner `.

File: projects/lesson-5/runtime/src/kitties.rs [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+        let kitty_id = Self::kitties_count();
+        if kitty_id == T::KittyIndex::max_value() {
+            return Err("Kitties count overflow");
+        }
+        Ok(kitty_id)
+    }
+
+    fn insert_owned_kitty(owner: &T::AccountId, kitty_id: T::KittyIndex) {
+        // 作业：调用 OwnedKitties::append 完成实现
+        <OwnedKitties<T>>::append(owner,kitty_id);
```
Comment (kaichaosun) [inline projects/lesson-5/runtime/src/kitties.rs]: You need to update kitty owner:
`<KittyOwner<T>>::insert(kitty_id, owner.clone());`

**Pull request merged**